source/codegen/templates/service_helpers.mako

```diff
@@ -187,11 +209,55 @@ ${initialize_enum_input_param(function_name, parameter)}
 ${initialize_len_input_param(parameter)}
 % elif common_helpers.is_pass_null_parameter(parameter):
 ${initialize_pass_null_param(parameter)}
+% elif common_helpers.is_varargs_parameter(parameter):
+${initialize_varargs_param(parameter)}
 % else:
 ${initialize_standard_input_param(function_name, parameter)}
 % endif
 </%def>
 
+<%def name="initialize_varargs_param(parameter)">\
+<%
+  config = data['config']
+  parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
+  field_name = common_helpers.camel_to_snake(parameter["name"])
+  stripped_grpc_type = common_helpers.strip_repeated_from_grpc_type(parameter['grpc_type'])
+  request_snippet = f'request->{field_name}()'
+  c_type = parameter['type']
+  c_type_pointer = c_type.replace('[]','*')
+  max_vector_size = parameter['max_length']
+%>\
+      auto ${parameter_name} = request->${field_name}();
+      if (${parameter_name}.size() == 0) {
+            return ::grpc::Status(::grpc::INVALID_ARGUMENT, "No values for ${parameter["name"]} were specified");
+      }
+      if (${parameter_name}.size() > ${max_vector_size}) {
+            return ::grpc::Status(::grpc::INVALID_ARGUMENT, "More than ${max_vector_size} values for ${parameter["name"]} were specified");
+      }
+% for member in parameter['varargs_type']['fields']:
+<%
+  member_c_type = member['type']
+  member_c_type_pointer = member_c_type.replace('[]','*')
+  # In gRPC fields, the names of the fields in the struct are lowercase
+  member_name = member['name'].lower()
+%>\
+      auto get_${member['name']}_if = [](const google::protobuf::RepeatedPtrField<${stripped_grpc_type}>& vector, int n) -> ${member_c_type_pointer} {
+            if (vector.size() > n) {
+% if common_helpers.is_string_arg(member):
```

Comment:
Since these methods don't use the standard initiailze parameter methods, they're not going to work for more interesting types (enums, timestamps,). I think that's fine, and I'm not sure how easy that would be to fix. Just feeling out the boundaries of what the "varargs" convention supports.